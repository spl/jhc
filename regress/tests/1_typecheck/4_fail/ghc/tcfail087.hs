{-# LANGUAGE UnboxedTuples #-}

-- !!! Check that unboxed tuples can't be function arguments
module ShouldFail where

data Ex = Ex (# Int,Int #)

f :: (# Int,Int #) -> Int
f x = error "urk"

g (# x,y #) = x


