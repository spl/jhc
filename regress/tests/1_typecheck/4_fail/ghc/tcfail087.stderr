
tcfail087.hs:6:14:
    Kind mis-match
    Expected kind `ArgKind', but `(# Int, Int #)' has kind `(#)'
    In the type `(# Int, Int #)'
    In the definition of data constructor `Ex'
    In the data type declaration for `Ex'
